libs/wingc/src/parser.rs

```diff
@@ -535,54 +539,62 @@ impl<'s> Parser<'s> {
 							)
 							.err();
 					}
-					if !is_resource && is_inflight {
+					if class_phase == Phase::Independent && is_inflight {
```

Comment:
Not sure I understand how the code could reach a point where the class phase is independent, since the grammar only supports defining preflight classes and inflight classes at the moment